OpenLaTeXStudio/EditorModule/src/latexstudio/editor/menu/SaveFile.java

```diff
@@ -45,6 +46,10 @@ public void actionPerformed(ActionEvent e) {
         String content = etc.getEditorContent();
         File file = etc.getCurrentFile();
         if (file == null) file = FileChooserService.getSelectedFile("tex", "TeX files", DialogType.SAVE, true);
+        
+        //Confirm if to overwrite an exsiting file.
+        file = SaveConfirmDialog.confirmFileSave(JOptionPane.NO_OPTION, file, etc.getCurrentFile());

```

Comment:
I don't like that we must first call "getSelectedFile" and then additionaly "confirmFileSave". Could we have a method that merges the logic of the two above? It could be named "getFileWithConfirmation".